rocAL/rocAL/CMakeLists.txt

```diff
@@ -31,15 +31,46 @@ set(BUILD_RALI true)
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/../../amd_openvx/cmake)
 
-if(GPU_SUPPORT AND "${BACKEND}" STREQUAL "OPENCL")
+if(GPU_SUPPORT)
     find_package(OpenCL QUIET)
     find_package(AMDRPP QUIET)
     find_package(TurboJpeg QUIET)
     find_package(Protobuf QUIET)
     # If FFMPEG is found, vx_amd_media is expected to be built as well
     find_package(FFmpeg QUIET)
     find_package(OpenCV QUIET)
-elseif(NOT GPU_SUPPORT OR "${BACKEND}" STREQUAL "HIP")
+endif()
+
+if( GPU_SUPPORT AND "${BACKEND}" STREQUAL "HIP")
+    if(NOT DEFINED HIP_PATH)
+        if(NOT DEFINED ENV{HIP_PATH})
+            set(HIP_PATH ${ROCM_PATH}/hip CACHE PATH "Path to which HIP has been installed")
+        else()
+            set(HIP_PATH $ENV{HIP_PATH} CACHE PATH "Path to which HIP has been installed")
+        endif()
+    endif()
+    list(APPEND CMAKE_MODULE_PATH ${HIP_PATH}/cmake)
+    # HSA_PATH
+    IF(NOT DEFINED ENV{HSA_PATH})
+        SET(HSA_PATH ${ROCM_PATH}/hsa)
+    ELSE()
+        SET(HSA_PATH $ENV{HSA_PATH})
+    ENDIF()
+    find_package(HIP QUIET REQUIRED)
+    if(HIP_FOUND)
+        SET(BUILD_RALI true)
+        message(STATUS "Found HIP for RALI: " ${HIP_VERSION} ${HIP_PATH} ${HCC_HOME})
+        if(HIP_COMPILER STREQUAL clang)
+            set(hip_library_name amdhip64)
+        else()
+            set(hip_library_name hip_hcc)
+        endif()
+        message(STATUS "HIP library set to : " ${hip_library_name})
+    else()
+        SET(BUILD_RALI false)
+        message(FATAL_ERROR "Could'nt find HIP. Ensure that HIP is either installed in /opt/rocm/hip or the variable HIP_PATH is set.")
+    endif()
+elseif(NOT GPU_SUPPORT)
```

Comment:
Move this to the else statement for line #34

Reply:
Done.